nupic/research/frameworks/dynamic_sparse/models/main.py

```diff
@@ -225,36 +226,51 @@ class SparseModel(BaseModel):
     def setup(self):
         super(SparseModel, self).setup()
 
+        # calculate sparsity masks
+        self.masks = []
+        self.num_params = []  # added for paper implementation
+
+        # define sparse modules
+        self.sparse_modules = []
+        for m in list(self.network.modules()):
+            if isinstance(m, DynamicSparseBase):
+                # TODO: Could be more fitting to name this 'dynamic_sparse_modules'
```

Comment:
If so, would be better to rename now.

Reply:
The models use this attribute in a lot of places, and Lucas has been working on many changes to the models as well. I can change it now, but I figured it may help to wait to avoid as many merge conflicts as there will be.

Reply:
On second thought. I think `sparse_modules` is fitting because the list takes no assumptions on whether the modules are dynamic+sparse or just sparse.

Reply:
ok.

Reply:
Please remove the TODO then! 😄

Reply:
You can disregard my changes, anyhow I will have to manually merge them later. Go ahead and do what you think it is best to fit this new architecture

Reply:
And as you said, sparse modules is fitting. In my current fork I actually have a second list called toprune_modules, which only have the modules being pruned